app/controllers/v1/events_controller.rb

```diff
@@ -0,0 +1,38 @@
+class V1::EventsController < ApplicationController
+  before_action :set_event, only: [:show, :update, :destroy]
+
+  def index
+    @events = Event.where(owner: @current_user)
+  end
+
+  def show
+  end
+
+  def create
+    @event = Event.new(event_params)
+    @event.save
+    render json: @event
+  end
+
+  def update
+    @event.update(event_params)
+    render json: @event
+  end
+
+  def destroy
+    @event.destroy
+    head :no_content
+  end
+
+  private
+
+  def set_event
+    @event = Event.find_by(id: params[:id], owner: @current_user)
+  rescue Mongoid::Errors::DocumentNotFound
+    head :unauthorized
+  end
```

Comment:
`DocumentNotFound` exception could be raised due two reasons:
1) There is no event found by `params[:id]`.  
2) The event exists, but `@current_user` is not the owner.

In the second case it's an unauthorized action, but in the first case... i think a `:not_found` must be returned.

What about this?
```ruby
  def set_event
    event = Event.find(params[:id])
    event.owner == @current_user ? @event = event : head(:unauthorized)
  rescue Mongoid::Errors::DocumentNotFound
    head :not_found
  end
```

Adding new test (I don't like introducing a magic number as ID, any idea?)
```ruby
  test 'return a not found status when try to show an nonexistent event' do
    get :show, { id: 20 }.merge!(@auth_params)
    assert_response :not_found
  end
```